simcon_project/conversation_templates/views/create_assignment.py

```diff
@@ -0,0 +1,142 @@
+from django.shortcuts import render
+from users.models import SubjectLabel, Assignment, Student, Researcher
+from conversation_templates.models import ConversationTemplate
+from django.views.generic import TemplateView
+from django.core import serializers
+from django.http import HttpResponse
+from django.core.mail import send_mail
+from apscheduler.schedulers.background import BackgroundScheduler
+from tzlocal import get_localzone
+import json
+import datetime
+
+
+class CreateAssignmentView(TemplateView):
+    def get(self, request):
+        stud=Student.objects.all().values('email', 'first_name','last_name','is_active')
+        label=SubjectLabel.objects.all()
+        template=ConversationTemplate.objects.all()
+        researchers=Researcher.objects.all().values('email')
+
+        student=json.dumps(list(stud))
+        researcher=json.dumps(list(researchers))
+        subLabel = serializers.serialize("json",label)
+        template = serializers.serialize("json",template)
+
+        return render(request, 'create_assignment.html', {
+            'student': student,
+            'subLabel': subLabel,
+            'template': template,
+            'researcher': researcher
+        })
+
+#Transfer string type list to list type
+def decode(str):
+    if str[0]!='[':
+        print("Decode failed. The data is not a list type.")
+        return
+    temp=str[1:-1]
+    temp=temp.split(',')
+    listTmp=[]
+    for t in temp:
+        t=t[1:-1]
+        listTmp.append(t)
+    return listTmp
+
+def sendMail(subject, msg, recipient, researcher):
+    send_mail(subject,msg,researcher,recipient,fail_silently=False)
+
+#Determine if data is empty.
+def isNull(data):
+    if len(data)==0 or (len(data)==1 and data[0]==''):
+        return True
+    return False
+
+def add_data(request):
+    #Error signs and error messages.
+    errMsg=''
+    success=0
+
+    #Get raw data
+    data=request.POST
+    name=data.get('name')
+    date=data.get('date')
+    researcher=request.user
+    #researcher='researcher@111.com'
+    students=data.get('stuData')
+    templates=data.get('tempData')
+    labels=data.get('labelData')
+
+    #Transfer string type list to list type
+    students=decode(students)
+    templates=decode(templates)
+    labels=decode(labels)
+
+    #Verify date
+    today=datetime.datetime.combine(datetime.date.today(), datetime.time(0, 0))
+    dateTmp=datetime.datetime.strptime(date, "%Y-%m-%d")
+    if dateTmp<=today:
+        success=1
+        errMsg=errMsg+'The assigned date cannot be today or before today.\n\n'
+
+    #Save the data that can be directly assigned to assignment.
+    assignment=Assignment()
+    assignment.name=name
+    assignment.date_assigned=date
+    researcher=Researcher.objects.get(email=researcher)
+    assignment.researcher=researcher
+    assignment.save()
```

Comment:
You are saving the assignment even though you have an error that says "the assigned date cannot be today or before today." Make sure to not save the assignment if it doesn't have a valid date.

Reply:
I added a judgment at the end of the program. If there is any error during the creation of the assignment, delete the assignment at the end.